ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayloadV3.java

```diff
@@ -58,73 +59,92 @@ public String getName() {
   }
 
   @Override
-  protected Optional<JsonRpcResponse> validateForkSupported(
+  protected ValidationResult<NewPayloadValidationResult> validateForkSupported(
       final Object reqId, final EnginePayloadParameter payloadParameter) {
     var cancun = timestampSchedule.hardforkFor(s -> s.fork().name().equalsIgnoreCase("Cancun"));
 
     if (cancun.isPresent() && payloadParameter.getTimestamp() >= cancun.get().milestone()) {
       if (payloadParameter.getDataGasUsed() == null
           || payloadParameter.getExcessDataGas() == null) {
-        return Optional.of(new JsonRpcErrorResponse(reqId, INVALID_PARAMS));
+        return ValidationResult.invalid(NewPayloadValidationResult.INVALID_NEW_PAYLOAD_PARAMS);
       }
     } else {
       if (payloadParameter.getDataGasUsed() != null
           || payloadParameter.getExcessDataGas() != null) {
-        return Optional.of(new JsonRpcErrorResponse(reqId, UNSUPPORTED_FORK));
+        return ValidationResult.invalid(NewPayloadValidationResult.INVALID_NEW_PAYLOAD_PARAMS);
```

Comment:
There's been a semantic change from UNSUPPORTED_FORK to INVALID_NEW_PAYLOAD_PARAMS here? Just checking it's not copy/paste error.